gr_modtool: c++ blocks: enabling C++ QA gen leads to broken lib/CMakeLists.txt

### What happened?

investigating #6960, I found that with noblock, sync, hier and  probably all other block types, gr_modtool breaks CMakeLists.txt

### System Information

OS: Linux, source build

### GNU Radio Version

3.11-git (main)

### Specific Version

v3.11.0.0git-649-gd7cdf222

### Steps to Reproduce the Problem

```shell
#!/bin/sh
modname="foobar"
blockname="nothing"
blocktype="hier"
rm -r "gr-${modname}"
gr_modtool newmod "${modname}"
cd gr-"${modname}"
gr_modtool add --add-cpp-qa --block-type "${blocktype}" --copyright dontcare --lang cpp "${blockname}"
mkdir build
cd build

cmake ..
```

The offending lines in lib/CMakeLists.txt are:

```cmake
  50   │ ########################################################################
  51   │ # Build and register unit test
  52   │ ########################################################################
  53   │ include(GrTest)
  54   │
  55   │ # If your unit tests require special include paths, add them here
  56   │ #include_directories()
  57   │ # List all files that contain Boost.UTF unit tests here
  58   │ list(APPEND test_foobar_sources)
  59   │ qa_nothing.cc
  60   │ # Anything we need to link to for the unit tests go here
  61   │ list(APPEND GR_TEST_TARGET_DEPS gnuradio-foobar)
  62   │
  63   │ if(NOT test_foobar_sources)
  64   │     message(STATUS "No C++ unit tests... skipping")
  65   │     return()
  66   │ endif(NOT test_foobar_sources)
  67   │
  68   │ foreach(qa_file ${test_foobar_sources})
  69   │     gr_add_cpp_test("foobar_${qa_file}" ${CMAKE_CURRENT_SOURCE_DIR}/${qa_file})
  70   │ endforeach(qa_file)
```


### Relevant log output

```shell
-- Found Doxygen: /usr/bin/doxygen (found version "1.9.6") found components: doxygen dot
CMake Error at lib/CMakeLists.txt:59:
  Parse error.  Expected a command name, got unquoted argument with text
  "qa_nothing.cc".


-- Configuring incomplete, errors occurred!
```

--- a/gr-utils/modtool/tools/util_functions.py
+++ b/gr-utils/modtool/tools/util_functions.py
@@ -9,9 +9,8 @@
 """ Utility functions for gr_modtool """
 
 
-import re
-import sys
 import os
+import re
 try:
     import readline
     have_readline = True
@@ -21,10 +20,17 @@ except ImportError:
 # None of these must depend on other modtool stuff!
 
 
-def append_re_line_sequence(filename, linepattern, newline):
-    """ Detects the re 'linepattern' in the file. After its last occurrence,
+def append_re_line_sequence(filename: str, linepattern: str, newline: str, closing_parentheses: str = ')') -> None:
+    """
+    Detects the re 'linepattern' in the file. After its last occurrence (and any other identical occurrences),
     paste 'newline'. If the pattern does not exist, append the new line
-    to the file. Then, write. """
+    to the file.
+
+    If 'closing_parentheses' is not None, e.g. ")", remove it from end of the matched line, paste the 'newline',
+    followed by a line containing the 'closing_parentheses'.
+
+    Finally, write the modified file.
+    """
     with open(filename, 'r') as f:
         oldfile = f.read()
     lines = re.findall(linepattern, oldfile, flags=re.MULTILINE)
@@ -33,7 +39,11 @@ def append_re_line_sequence(filename, linepattern, newline):
             f.write(newline)
         return
     last_line = lines[-1]
-    newfile = oldfile.replace(last_line, last_line + newline + '\n')
+    if closing_parentheses is not None and last_line.rstrip().endswith(closing_parentheses):
+        modified_last = f'\n{last_line.rstrip()[:-len(closing_parentheses)]}\n{closing_parentheses}\n'
+    else:
+        modified_last = last_line + '\n'
+    newfile = oldfile.replace(last_line, modified_last)
     with open(filename, 'w') as f:
         f.write(newfile)
 

